doc/source/reference/c-api/iterator.rst

```diff
@@ -203,6 +203,69 @@ is used to control the memory layout of the allocated result, typically
     }
 
 
+Multi Index Tracking Example
+----------------------------
+
+This example shows you how to work with the :c:data:`NPY_ITER_MULTI_INDEX` flag. For simplicity, we assume the argument is a two-dimensional array.
+
+.. code-block:: c
+
+   void PrintMultiIndex(PyArrayObject *arr) {
```

Comment:
```suggestion
   int
   PrintMultiIndex(PyArrayObject *arr) {
```
And alwasy return `-1` on error and `0` on success.  Examples should take care to propagate errors.